frontend/packages/console-app/src/components/nav/NavHeader.tsx

```diff
@@ -119,6 +120,7 @@ const NavHeader: React.FC<NavHeaderProps> = ({ onPerspectiveSelected }) => {
       setClusterDropdownOpen(false);
       setActiveCluster(cluster);
       // TODO: Move this logic into `setActiveCluster`?
+      startGQLClient();
       dispatch(clearSSARFlags());
       dispatch(detectFeatures());
```

Comment:
I think we should create a function which will do all the necessary dispatches as these 3 function calls are repeating in couple more places.

Also I think we should re-run API discovery

Reply:
Feel free to add the api discovery here. We only left it out because we knew it wouldn't work yet :)

Reply:
I will create a separate PR for that. There is some caching involved on frontend side so Im not sure that running just `dispatch(startAPIDiscovery())` will be correct.